### PR TITLE
45 enable tracking of blockchain txs

### LINKED ISSUE #45: Enable tracking of blockchain txs
**Is your feature request related to a problem? Please describe.**
Track did operation blockchain txs in storage

### PR DESCRIPTION
**Pull request by Essbante**

### CODE REVIEW

File: src/main/kotlin/com/rootsid/wal/library/mongoimpl/BlockchainTxLogDocStorage.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.mongodb.client.MongoDatabase
+import com.rootsid.wal.library.mongoimpl.config.DefaultMongoDbConn
+import com.rootsid.wal.library.mongoimpl.document.BlockchainTxLogDocument
+import com.rootsid.wal.library.wallet.model.BlockchainTxAction
+import com.rootsid.wal.library.wallet.model.BlockchainTxLog
+import com.rootsid.wal.library.wallet.storage.BlockchainTxLogStorage
+import org.litote.kmongo.*
+import java.util.*
+
+class BlockchainTxLogDocStorage(db: MongoDatabase? = null, collectionName: String = "txLog") : BlockchainTxLogStorage {
```
Comment (rogelio-blanco) [inline src/main/kotlin/com/rootsid/wal/library/mongoimpl/BlockchainTxLogDocStorage.kt]: change the name of the collection to plural, like tx_logs

File: src/main/kotlin/com/rootsid/wal/library/mongoimpl/document/BlockchainTxLogDocument.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.rootsid.wal.library.mongoimpl.document
+
+import com.rootsid.wal.library.wallet.model.BlockchainTxAction
+import com.rootsid.wal.library.wallet.model.BlockchainTxLog
+import io.iohk.atala.prism.api.models.AtalaOperationStatus
+import io.iohk.atala.prism.api.models.AtalaOperationStatusEnum
+import kotlinx.serialization.Serializable
+
+@Serializable
```
Comment (rogelio-blanco) [inline src/main/kotlin/com/rootsid/wal/library/mongoimpl/document/BlockchainTxLogDocument.kt]: Remove this and make the BlockchainTxLog implement Serializable

File: src/main/kotlin/com/rootsid/wal/library/wallet/storage/BlockchainTxLogStorage.kt [hunk 3]
```diff
@@ -1,9 +1,22 @@
 package com.rootsid.wal.library.wallet.storage
 
+import com.rootsid.wal.library.wallet.model.BlockchainTxAction
 import com.rootsid.wal.library.wallet.model.BlockchainTxLog
+import java.util.*
 
 interface BlockchainTxLogStorage {
```
Comment (rogelio-blanco) [inline src/main/kotlin/com/rootsid/wal/library/wallet/storage/BlockchainTxLogStorage.kt:7]: Make this implement Serializable

**Review by rogelio-blanco — APPROVED** [review]

File: src/main/kotlin/com/rootsid/wal/library/mongoimpl/BlockchainTxLogDocStorage.kt (hunk 1 above)
Comment (Essbante) [reply src/main/kotlin/com/rootsid/wal/library/mongoimpl/BlockchainTxLogDocStorage.kt]: ok

File: src/main/kotlin/com/rootsid/wal/library/mongoimpl/document/BlockchainTxLogDocument.kt (hunk 2 above)
Comment (Essbante) [reply src/main/kotlin/com/rootsid/wal/library/mongoimpl/document/BlockchainTxLogDocument.kt]: ok

File: src/main/kotlin/com/rootsid/wal/library/wallet/storage/BlockchainTxLogStorage.kt (hunk 3 above)
Comment (Essbante) [reply src/main/kotlin/com/rootsid/wal/library/wallet/storage/BlockchainTxLogStorage.kt:7]: ok

**Pull request merged**